### PR TITLE
ROI Implementation

### PR DESCRIPTION
**Pull request by thedevleon**
This is a PR for the implementation of ROI (see https://github.com/PX4/Firmware/issues/4584).
The gimbal driver will be implemented at a later stage (and a separate PR)

#### Progress:
- [x] Implement the `MAV_ROI` enum in the `vehicle_command` topic
- [x] Create a new uORB topic for the vehicle ROI (named `vehicle_roi`)
- [x] Implement the parsing of `MAV_CMD_NAV_ROI` and `MAV_CMD_DO_SET_ROI` in `commander.cpp` and the advertising of the ROI on the `vehicle_roi` topic
- [x] Rename the current gimbal driver to something else like RC_Gimbal

### CODE REVIEW

File: src/drivers/rc_gimbal/rc_gimbal_params.c [hunk 1]
```diff
@@ -51,7 +51,7 @@
  * @boolean
  * @group Gimbal
```
Comment (dagar) [inline src/drivers/rc_gimbal/rc_gimbal_params.c]: Change the param group too?

File: msg/vehicle_command.msg [hunk 2]
```diff
@@ -2,63 +2,64 @@
 uint32 VEHICLE_CMD_CUSTOM_0 = 0				# test command
 uint32 VEHICLE_CMD_CUSTOM_1 = 1				# test command
 uint32 VEHICLE_CMD_CUSTOM_2 = 2				# test command
-uint32 VEHICLE_CMD_NAV_WAYPOINT = 16			# Navigate to MISSION. |Hold time in decimal seconds. (ignored by fixed wing, time to stay at MISSION for rotary wing)| Acceptance radius in meters (if the sphere with this radius is hit, the MISSION counts as reached)| 0 to pass through the WP, if > 0 radius in meters to pass by WP. Positive value for clockwise orbit, negative value for counter-clockwise orbit. Allows trajectory control.| Desired yaw angle at MISSION (rotary wing)| Latitude| Longitude| Altitude|
```
Comment (LorenzMeier) [inline msg/vehicle_command.msg]: What is going on here - did you fix whitespace or why is the diff so large?

Comment (thedevleon) [inline msg/vehicle_command.msg]: I'm using atom to edit everything, seemed like it automatically adjusted the whitespaces, yes. Is this okay? I can undo it if you want.

Comment (LorenzMeier) [inline msg/vehicle_command.msg]: No, that's fine. I just needed to understand the background. In general we try to reduce whitespace as much as possible.

**Comment by thedevleon** [comment]
This should be good to go, please confirm. I can do some testing once I have implemented the mount (gimbal) driver.

**Comment by thedevleon** [comment]
Superseded by https://github.com/PX4/Firmware/pull/4624.

**Pull request closed without merging**